src/geocat/f2py/linint2_wrapper.py

```diff
@@ -98,23 +98,33 @@ def linint1(fi, xo, xi=None, icycx=0, msg_py=None):
     # ''' signature : fo = dlinint1(xi,fi,xo,[icycx,xmsg,iopt])
 
     # ''' Start of boilerplate
-    if not isinstance(fi, xr.DataArray):
-        if (xi is None):
-            raise CoordinateError(
-                "linint2: Argument xi must be provided explicitly unless fi is an xarray.DataArray."
-            )
-
-        fi = xr.DataArray(fi,)
-        fi_chunk = dict([(k, v) for (k, v) in zip(list(fi.dims), list(fi.shape))
-                        ])
-
-        fi = xr.DataArray(
-            fi.data,
-            coords={
-                fi.dims[-1]: xi,
-            },
-            dims=fi.dims,
-        ).chunk(fi_chunk)
+
+    # Check input data type
+    check(xo, data_type=np.ndarray)
+    check(yo, data_type=np.ndarray)
+    if (xi is not None) | (yi is not None):
+        check(xi, data_type=np.ndarray)
+        check(yi, data_type=np.ndarray)
+
+    # Check if fi is xarray.DataArray, if not change it
+    if check(fi, is_xarray=False) == True:
```

Comment:
I agree. When writing this line, I wanted to make sure that by calling ```check(fi, is_xarray=False)``` there wouldn't be an issue if user used```check(fi, is_xarray=True)```. However, something I wanted to double check is the necessity of having ```is_xarray``` as a True/False statement. It would seem that we would only need it to be defined for one option (only True or only False) and then have the wrapper function defined for that (e.g. only using ```check(fi, is_xarray=True)``` or ```check(fi, is_xarray=False)``` in the wrapper). That being said, I think this may also need to be reformatted in order to resolve your second comment above, because it is true that the wrapper functions can handle both the xarray.DataArray and numpy.ndarray. I believe that Austin and I used ```is_xarray``` due to the original ncomp wrapper check saying that the variable type should be in is xarray.DataArray only. 

With respect to your first, third, and fourth point above, and considering the other changes requested for this PR, I think we should re-visit this PR as a team to make sure it is going in the right direction for this repository. In the mean time, I will work on the changes outlined below, and ensuring they work with the two wrappers I have been testing implementation in.

Reply:
> With respect to your first, third, and fourth point above, and considering the other changes requested for this PR, I think we should re-visit this PR as a team to make sure it is going in the right direction for this repository. In the mean time, I will work on the changes outlined below, and ensuring they work with the two wrappers I have been testing implementation in.

I agree. Let us discuss these in our weekly meeting, and in the meantime, I will make a document to write down major points that would need discussion/modification etc.